builtin/providers/aws/resource_aws_autoscaling_group.go

```diff
@@ -287,7 +287,12 @@ func resourceAwsAutoscalingGroupDelete(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
-	return nil
+	return resource.Retry(5*time.Minute, func() error {
+		if g, _ = getAwsAutoscalingGroup(d, meta); g != nil {
+			return fmt.Errorf("Auto Scaling Group still exists")
+		}
```

Comment:
`getAwsAutoscalingGroup` only returns a non-nil `g` if it found the ASG, in which case we want to continue to pool